### PR TITLE
email parse depends on python patch version

### PR DESCRIPTION
**Pull request by austinweisgrau**
The behavior of email.parseaddr depends on the python patch version.

See https://github.com/python/cpython/issues/102988 or associated changelogs, e.g. https://docs.python.org/3.8/whatsnew/changelog.html#python-3-8-20-final

### CODE REVIEW

**Comment by bmos** [comment]
Great work tracking down the changelog that describes this behavior!

**Review by shaunagm — APPROVED** [review]
Thanks so much for tracking this down! It was a tricky one. I like your solution to add a condition to the tests, rather than require folks to not pin to a specific python version.

**Comment by shaunagm** [comment]
Although, question: how would this create the pattern of test failures we saw, where tests were failing on Mac and Windows up to 3.12 (but not 3.12) and Ubuntu on 3.8 (but not 3.9-3.12)?

**Comment by austinweisgrau** [comment]
There isn't actually a breaking change in this PR, rather the change in behavior of email.parseaddr is with the python minor patch version which is very upstream of us.

I believe the the test failures were due to different times that the patches were merged into different python versions. The change in parseaddr was merged into each python major version in different releases at presumably different times.

**Comment by shaunagm** [comment]
Sorry, that was a bit of inside baseball - I'm cutting the next release soon and I always check the PRs for breaking changes. When I see this PR, I'll remember to add a small note to the release notes calling out this fix in particular, just in case it's affected anyone else. We had a hard enough time detecting that it was due to a python patch version, so if a Parsons user runs into it I'd hope they could see it in the Parsons changelog notes instead of having to delve even further. 

tl;dr I won't actually list it as a breaking change.

**Comment by shaunagm** [comment]
> I believe the different test failures were due to different times that the patches were merged into different python versions. The change in parseaddr was merged into each python major version in different releases at presumably different times.

Also this explanation makes sense, thank you!
